src/Microsoft.SqlTools.ServiceLayer/Scripting/Scripter.cs

```diff
@@ -31,218 +32,229 @@ private void Initialize()
             AddSupportedType(DeclarationType.Synonym, GetSynonymScripts, "Synonym", "");
             AddSupportedType(DeclarationType.ScalarValuedFunction, GetScalarValuedFunctionScripts, "Function", "scalar-valued function");
             AddSupportedType(DeclarationType.TableValuedFunction, GetTableValuedFunctionScripts, "Function", "table-valued function");
-        }
+        }
 
         /// <summary>
         /// Script a Table using SMO
         /// </summary>
         /// <param name="objectName">Table name</param>
         /// <param name="schemaName">Schema name</param>
         /// <returns>String collection of scripts</returns>
-        internal StringCollection GetTableScripts(string objectName, string schemaName, ScriptingOptions scriptingOptions = null)
+        internal bool GetTableScripts(string objectName, string schemaName, string filePath)
         {
             try
             {
-                Table smoObject = string.IsNullOrEmpty(schemaName) ? new Table(this.Database, objectName) : new Table(this.Database, objectName, schemaName);
-                smoObject.Refresh();
-                return (scriptingOptions == null) ? smoObject.Script() : smoObject.Script(scriptingOptions);
+                ScriptingParams requestParams = new ScriptingParams
+                {
+                    FilePath = filePath,
+                    ConnectionString = this.connectionInfo.OwnerUri,
+                    IncludeTypes = {"Table"},
+                    ScriptOptions = new ScriptOptions
+                    {
+                        TypeOfDataToScript = "SchemaOnly",
+                        AppendToFile = true
+                    },
+                };
+                ScriptingScriptOperation operation = new ScriptingScriptOperation(requestParams);
+                operation.Execute();
+                return true;
             }
             catch (Exception ex)
             {
                 Logger.Write(LogLevel.Error,"Exception at PeekDefinition GetTableScripts : " + ex.Message);
-                return null;
+                return false;
             }
-        }
+        }
 
         /// <summary>
         /// Script a View using SMO
         /// </summary>
         /// <param name="objectName">View name</param>
         /// <param name="schemaName">Schema name</param>
         /// <returns>String collection of scripts</returns>
-        internal StringCollection GetViewScripts(string objectName, string schemaName, ScriptingOptions scriptingOptions = null)
+        internal bool GetViewScripts(string objectName, string schemaName, string filePath)
         {
             try
             {
                 View smoObject = string.IsNullOrEmpty(schemaName) ? new View(this.Database, objectName) : new View(this.Database, objectName, schemaName);
                 smoObject.Refresh();
-                return (scriptingOptions == null) ? smoObject.Script() : smoObject.Script(scriptingOptions);
+                return true;

```

Comment:
>true [](start = 23, length = 4)

don't we need to call some methods to actually generate the view script?

Reply:
This was my bad, a few files for the peek definition changes slid into this PR. This whole file will be deleted.